internal/broker/config_test.go

```diff
@@ -150,3 +150,242 @@ func TestParseConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestParseUserConfig(t *testing.T) {
+	t.Parallel()
+	tests := map[string]struct {
+		config string
+		check  func(userConfig)
+		err    error
+	}{
+		"All_are_allowed": {
+			config: `
+[oidc]
+issuer = https://issuer.url.com
+client_id = client_id
+
+[users]
+allowed_users = ALL
+owner = machine_owner
+home_base_dir = /home
+allowed_ssh_suffixes = @issuer.url.com
+			`,
```

Comment:
For consistency, lets extract the config that's used in the test cases into a map, like it's done [here](https://github.com/nsklikas/authd-oidc-brokers/blob/90573fc10b23e7435dee63b41d73d74cc9efcd3b/internal/broker/config_test.go#L16).